remove the use of deleted default ctor of ValidatorKeys

ckeshava opened the pull request:
The default constructors of `PublicKey, Account, ValidatorKeys` classes have been removed in a recent release of rippled codebase. This PR performs a similar change in this tools codebase.

ckeshava commented:
The only commit that is relevant to this PR is e4ac958642a9f81782884de1e7e97c63fb317a82
I have merged John's PR hoping to trigger Github CI Actions, but I don't observe any CI runs

thejohnfreeman commented on src/ValidatorKeys.cpp:
```diff
@@ -46,9 +46,9 @@ ValidatorKeys::ValidatorKeys (KeyType const& keyType)
     : keyType_ (keyType)
     , tokenSequence_ (0)
     , revoked_ (false)
+    , publicKey_ (generateKeyPair (keyType_, randomSeed ()).first)
+    , secretKey_ (generateKeyPair (keyType_, randomSeed ()).second)
```
I think you should put the return value of `generateKeyPair(keyType_, randomSeed())` into a default argument and then move the pieces in these member initializers.

ximinez commented on src/ValidatorKeys.cpp:
```diff
@@ -46,9 +46,9 @@ ValidatorKeys::ValidatorKeys (KeyType const& keyType)
     : keyType_ (keyType)
     , tokenSequence_ (0)
     , revoked_ (false)
+    , publicKey_ (generateKeyPair (keyType_, randomSeed ()).first)
+    , secretKey_ (generateKeyPair (keyType_, randomSeed ()).second)
```
This is going to give you two unrelated keys. A better way to ensure you're working with the same keypair is to add a `std::pair` member, initialize that first, then initialize pk and sk from that.
In the .h
```
std::pair<PublicKey, SecretKey> const keyPair_;
PublicKey const& publicKey_;
SecretKey const& secretKey_;
```
In the ctor
```
, keyPair_ (generateKeyPair (keyType_, randomSeed ()))
, publicKey_ (keyPair_.first)
, secretKey_ (keyPair_.second)
```

ckeshava replied:
I have done something slightly different. I feel my solution has lesser repetition of information. But I'm doing this blind, I've been able to run a local CI test ://

ximinez commented on src/ValidatorKeys.cpp:
```diff
@@ -60,8 +60,8 @@ ValidatorKeys::ValidatorKeys (
     , secretKey_ (secretKey)
     , tokenSequence_ (tokenSequence)
     , revoked_ (revoked)
+    , publicKey_ (derivePublicKey(keyType_, secretKey_))
```
And now because of the `keyPair_` added in the first comment, you're going to need to initialize these differently, too.
```
, keyPair_ (secretKey, derivePublicKey(keyType_, secretKey))
, secretKey_ (keyPair_.first)
, publicKey_ (keyPair_.second)
, tokenSequence_ (tokenSequence)
, revoked_ (revoked)
```
(Also initialize sk and pk together.)

thejohnfreeman commented:
There are no GitHub Actions to run.

ckeshava commented:
@thejohnfreeman  👍  I mistakenly thought that your PR had Github Actions changes, my bad

ckeshava commented:
thanks for the quick review john!

ckeshava commented:
This code builds correctly and all the unit tests pass. I've verified on MacOS

ckeshava commented:
@ximinez  can you please take one more look at this? The code looks correct from my perspective

ximinez commented on src/ValidatorKeys.h:
```diff
@@ -48,8 +48,18 @@ class ValidatorKeys
-    SecretKey secretKey_;
+
+    // struct used to contain both public and secret keys
+    struct Keys {
+      PublicKey publicKey;
+      SecretKey secretKey;
+
+      Keys() = delete;
+      Keys(std::pair<PublicKey, SecretKey> p)
+          : publicKey(p.first), secretKey(p.second) {}
+    };
+
+    Keys keys_;
```
Nice!

ximinez commented on src/ValidatorKeys.cpp:
```diff
@@ -274,7 +270,8 @@ ValidatorKeys::revoke ()
 {
-    return strHex(ripple::sign (publicKey_, secretKey_, makeSlice (data)));
+  return strHex(
+      ripple::sign(keys_.publicKey, keys_.secretKey, makeSlice(data)));
```
Nit: We don't have formatting enforced by `clang-format` in this project, but I noticed here in particular that the line starting `return` needs two more spaces to fit the expected formatting.

ximinez commented on src/ValidatorKeys.h:
```diff
@@ -48,8 +48,18 @@ class ValidatorKeys
+
+    // struct used to contain both public and secret keys
+    struct Keys {
+      PublicKey publicKey;
+      SecretKey secretKey;
+
+      Keys() = delete;
+      Keys(std::pair<PublicKey, SecretKey> p)
+          : publicKey(p.first), secretKey(p.second) {}
```
Nit: Formatting. Indent these lines by two more spaces

ximinez commented on src/ValidatorKeys.h:
```diff
@@ -79,14 +89,11 @@ class ValidatorKeys
-            keyType_ == rhs.keyType_ &&
-            tokenSequence_ == rhs.tokenSequence_ &&
-            publicKey_ == rhs.publicKey_;
+    inline bool operator==(ValidatorKeys const &rhs) const {
+      return revoked_ == rhs.revoked_ && keyType_ == rhs.keyType_ &&
+             tokenSequence_ == rhs.tokenSequence_ &&
+             keys_.publicKey == rhs.keys_.publicKey &&
+             keys_.secretKey == rhs.keys_.secretKey;
```
Nit: Formatting. The `return` needs two more spaces. And the other conditions probably will, too.

ximinez commented on src/ValidatorKeys.h:
```diff
@@ -124,11 +131,7 @@ class ValidatorKeys
-    {
-        return publicKey_;
-    }
+    PublicKey const &publicKey() const { return keys_.publicKey; }
```
Nit: Formatting. Please restore the formatting of the original function (other than that extra space between `publicKey` and `()`.

ximinez commented on src/ValidatorKeys.cpp:
```diff
@@ -42,25 +42,19 @@ ValidatorToken::toString () const
-}
+ValidatorKeys::ValidatorKeys(KeyType const &keyType)
+    : keyType_(keyType), tokenSequence_(0), revoked_(false),
+      keys_(generateKeyPair(keyType_, randomSeed())) {}
```
Nit: Formatting. Please restore the original formatting.

ximinez reviewed (approved):
This looks fine other than some issues with code formatting. 

I'm marking this as accepted because we can proceed without fixing the formatting, and open another PR to do everything all at once. But I would really like to at least see the places where the formatting was "ok" and changed to be restored.

ckeshava commented:
@ximinez  I have addressed your comments 👍 
I did execute this command: `git clang-format --extensions c,cpp,h,cxx,ipp` -> but it didn't make any changes to the files (although it acknowledged that `ValidatorKeys/.cpp/.h` files were modified)

ximinez commented:
> [...]but it didn't make any changes to the files[....]

Probably because there is no `.clang-format` file to tell it what to do.
